### PR TITLE
SGIFormer

### PR DESCRIPTION
**Pull request by Streakfull**
# In this PR,
* Adapted the  SGIFormer [repo](https://github.com/RayYoh/SGIFormer?tab=readme-ov-file), to the current pointcept repo.
* Added the transformations to obtain the superpoints for the scannetpp dataset.
* Added scannetpp config files for the isseg-files.

---
## Experimental results on scannetpp V2
image

---
*Note 1*: All credit goes to @RayYoh, and the original SGIFormer [repo](https://github.com/RayYoh/SGIFormer?tab=readme-ov-file).
*Note 2: this PR depends on  [PR 427](https://github.com/Pointcept/Pointcept/pull/428)*

### CODE REVIEW

**Comment by RongkunYang** [comment]
I process the scannet++ dataset use the sgiFormer to process Superpoint data

**Comment by RongkunYang** [comment]
I would like to consult another question, when I train the SGIFormer network in this code, the Loss will get NaN, which would damage the training process, could you help me about this?
Thank you.

**Comment by Streakfull** [comment]
> Hello, dear authors, I follow the code of this PR to process the scannet++ dataset, and I found that the color is processed all into 0, this will cause an issue.

Hello, can you please post the preprocessing steps you have done so far to help in debugging? Because the color should not all 0.

> I would like to consult another question, when I train the SGIFormer network in this code, the Loss will get NaN, which would damage the training process, could you help me about this? Thank you.

This didn't happen when we were training and you can find the logs on: :https://github.com/scannetpp/scannetpp?tab=readme-ov-file#benchmarks](https://github.com/scannetpp/scannetpp?tab=readme-ov-file#benchmarks). However, it might be related to the preprocessing issue you have mentioned, so lets solve the first issue first, and see this after.

**Comment by RongkunYang** [comment]
OK, thank you for your reply, I guess the NaN loss is caused by large batch size and distributed training, and I found the learning rate seems to be inconsistent in the config file(https://github.com/Streakfull/Pointceptog/blob/SpFormer/configs/scannetpp/insseg-sgiformer-spunet-v2.py), the maximum lr in schedule is different with that in param group optimizer (e.g. 0.003 and 0.0003), and the learning rate lr for main param group is 0.001, I'm not sure this is exact, since the SPFormer and following works usually set the lr to 1e-4.
image


Thank you very much.
